src/System.ValueTuple/src/System/ValueTuple/ValueTuple.cs

```diff
@@ -338,6 +338,7 @@ bool IStructuralEquatable.Equals(object other, IEqualityComparer comparer)
 
             var objTuple = (ValueTuple<T1>)other;
 
+            if (comparer == null) comparer = EqualityComparer<object>.Default;
             return comparer.Equals(Item1, objTuple.Item1);
```

Comment:
Or `object.Equals`, see https://github.com/dotnet/corefx/issues/18528#issuecomment-298748839
I'm not opposed.